### PR TITLE
Feature enhance start stop scripts

### PR DESCRIPTION
**Pull request by kvandra**
Added two delete scripts to provide automation for these tasks
Removed set -ex from scripts to minimize output
updated publish to check the status of gradlew publish command
enhanced start-job,sh to check different stages for creating the flink application - i.e. FlinkCluster created, FlinkApplication is deployed and checkpoint completed

tested with both data generator and stream-to-stream

### CODE REVIEW

File: scripts/delete-job.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Delete a Flink job.
+# It is recommend to start Flink jobs using the scripts in the jobs subdirectory.
+
+### set -ex
```
Comment (claudiofahey) [inline scripts/delete-job.sh]: Remove this commented command.

File: scripts/delete-pravega-stream.sh [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+### set -ex
+
+scope="edge"
+stream_name="sensor-stream"
+###ret_policy_type=d
```
Comment (claudiofahey) [inline scripts/delete-pravega-stream.sh]: Remove commented commands.

File: scripts/delete-pravega-stream.sh [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+### set -ex
+
+scope="edge"
+stream_name="sensor-stream"
```
Comment (claudiofahey) [inline scripts/delete-pravega-stream.sh]: It is not safe to have a default stream name to delete. This should be a required argument.

File: scripts/publish.sh [hunk 4]
```diff
@@ -9,12 +9,12 @@
 
 # Build and publish the Java JAR to the Maven repo on SDP.
 
-set -ex
+### set -ex
```
Comment (claudiofahey) [inline scripts/publish.sh]: `set -ex` sets two options. `-x` prints every command before executing this. This is very useful for troubleshooting but I'm ok with removing it. `-e` causes the script to abort if any command exits with an error. This needs to stay. So change this to `set -e`.

File: scripts/publish.sh [hunk 5]
```diff
@@ -36,6 +36,11 @@ else
     echo publish.sh: Building and publishing from source
     cd ${ROOT_DIR}
     ./gradlew publish ${GRADLE_OPTIONS}
+    echo "STATUS $?"
```
Comment (claudiofahey) [inline scripts/publish.sh]: No need to check for errors if you keep `set -e`.

File: scripts/start-job.sh [hunk 6]
```diff
@@ -44,4 +47,62 @@ helm upgrade --install --timeout 600s --debug --wait \
     --set "mavenCoordinate.version=${APP_VERSION}" \
     $@
 
-watch "kubectl get FlinkApplication -n ${NAMESPACE} ; kubectl get pod -o wide -n ${NAMESPACE}"
+printf "Waiting for FlinkCluster to start - should start in about 5 minutes\n"
+while : ; do
+    state=$(kubectl get FlinkCluster -n ${NAMESPACE} ${RELEASE_NAME} | grep ${RELEASE_NAME} | awk '{print $7}')
```
Comment (claudiofahey) [inline scripts/start-job.sh:52]: Very nice!

File: scripts/start-job.sh [hunk 7]
```diff
@@ -44,4 +47,62 @@ helm upgrade --install --timeout 600s --debug --wait \
...
+    if [[ "$state" == "Running" ]]; then
+        break;
+    fi
+done
```
Comment (claudiofahey) [inline scripts/start-job.sh:57]: You should add `sleep 1s` here so that it doesn't call kubectl too often.

**Review by claudiofahey — CHANGES_REQUESTED** [review]

File: scripts/delete-pravega-stream.sh (hunk 2 above)
Comment (kvandra) [reply scripts/delete-pravega-stream.sh]: removed comment

File: scripts/delete-pravega-stream.sh (hunk 3 above)
Comment (kvandra) [reply scripts/delete-pravega-stream.sh]: Made both scope and stream_name blank and added a test if either item is not specified on the command line

File: scripts/publish.sh (hunk 4 above)
Comment (kvandra) [reply scripts/publish.sh]: Changed to set -e

File: scripts/publish.sh (hunk 5 above)
Comment (kvandra) [reply scripts/publish.sh]: removed the echo statement

File: scripts/start-job.sh (hunk 6 above)
Comment (kvandra) [reply scripts/start-job.sh:52]: Thanks !!

File: scripts/start-job.sh (hunk 7 above)
Comment (kvandra) [reply scripts/start-job.sh:57]: added sleep 5s since it takes minutes for operation to complete

**Review by claudiofahey — APPROVED** [review]
Thanks for the PR!

**Pull request merged**